lib/engine/step/g_18_fl/token.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require_relative '../token'
+
+module Engine
+  module Step
+    module G18FL
+      class Token < Token
+        include Tokener
+        ACTIONS = %w[place_token hex_token pass].freeze
+
+        def actions(entity)
+          return [] unless entity == current_entity
+          return [] unless can_place_token?(entity)
+
+          ACTIONS
+        end
+
+        def can_place_token?(entity)
+          current_entity == entity &&
+            (tokens = available_tokens(entity)).any? &&
```

Comment:
!empty?

Reply:
Fixed

Also I did a regex search and I see `any?` used instead of `empty?` a fair bit throughout the codebase; would it be worth submitting a refactor PR that changes this for all base and alpha/beta/production games?